### PR TITLE
lobby

### LINKED ISSUE #51: Lobby screen
Initial screen to choose difficulty amount of players etc.

### PR DESCRIPTION
**Pull request by H1D**
- **added lobby screen closes #51**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+import { GameConfig } from "@/state/types";
 
-function App() {
-  return (
-    <div className="flex h-full w-full justify-center">
-      <GameBoard />
-    </div>
-  );
-}
+const App: React.FC = () => {
```
Comment (DreyerChris) [inline src/App.tsx:7]: Any specific reason for the typing here? Our other components dont look like this so I think we should stick to a standard

File: src/components/Croupier.tsx [hunk 2]
```diff
@@ -106,9 +107,22 @@ const Croupier = () => {
...
+              position={[-6, 10, 0]}
+              fontSize={5}
+              color="white"
+              shadowColor="white"
+              anchorX="left"
+              anchorY="bottom"
+              textAlign="left"
+            >
+              {/* removing wierd space due to emoji */}
+              {gameState.config.playerNames[index].replace(" ", "")}
```
Comment (DreyerChris) [inline src/components/Croupier.tsx:123]: Why not store the player name on the player itself?

File: src/components/Lobby/index.tsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+import { Select, SelectTrigger, SelectValue, SelectContent, SelectItem } from "@/components/ui/select";
+import { Popover, PopoverTrigger, PopoverContent } from "@radix-ui/react-popover";
+import { InfoIcon } from "@/components/ui/icons";
+import { GameConfig } from "@/state/types";
+
+interface LobbyScreenProps {
+  onStartGame: (settings: GameConfig) => void;
+}
+
+const LobbyScreen: React.FC<LobbyScreenProps> = ({ onStartGame }) => {
```
Comment (DreyerChris) [inline src/components/Lobby/index.tsx:17]: Same here, this is not the component standard that we've been using

File: src/hooks/useBloker.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+// useBlocker.ts
```
Comment (DreyerChris) [inline src/hooks/useBloker.ts:1]: typo in name of file

File: src/lib/utils.ts [hunk 5]
```diff
@@ -4,3 +4,137 @@ import { twMerge } from "tailwind-merge";
...
+  const characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+  let result = "";
+  for (let i = 0; i < length; i++) {
+    const randomChar = characters.charAt(Math.floor(Math.random() * characters.length));
+    result += randomChar;
+  }
+  return result;
+}
+
+export function generateRandomName(existingNames: Set<string>): string {
```
Comment (DreyerChris) [inline src/lib/utils.ts:18]: Haha nice

File: src/App.tsx (hunk 1 above)
Comment (H1D) [reply src/App.tsx:7]: not really. Just some ChatGPT artefacts =)

Comment (DreyerChris) [reply src/App.tsx]: 🤖

**Pull request merged**